include/language.php

```diff
@@ -148,78 +180,234 @@ class PLL_Language {
+	 * HTML markup of the custom flag if it exists.
 	 *
 	 * @var string|null
 	 */
 	public $custom_flag;
 
 	/**
-	 * Constructor: builds a language object given its two corresponding terms in 'language' and 'term_language' taxonomies.
+	 * Stores language term properties (like term IDs and counts) for each language taxonomy (`language`,
+	 * `term_language`, etc).
+	 * This stores the values of the properties `$term_id`, `$term_taxonomy_id`, `$count`, `$tl_term_id`,
+	 * `$tl_term_taxonomy_id`, and `$tl_count`.
+	 *
+	 * @var array[] Array keys are property names.
+	 *
+	 * @phpstan-var array<non-empty-string, array<non-empty-string, positive-int>>
+	 */
+	protected $term_props = array();
+
+	/**
+	 * Constructor: builds a language object given the corresponding data.
 	 *
 	 * @since 1.2
+	 * @since 3.4 Only accepts one argument.
+	 *
+	 * @param array $language_data {
+	 *     Language object properties stored as an array.
+	 *
+	 *     @type array[] $term_props      An array of language term properties. Array keys are language taxonomy names
+	 *                                    (`language` and `term_language` are mandatory), array values are arrays of
+	 *                                    language term properties (`term_id`, `term_taxonomy_id`, and `count`).
+	 *     @type string  $name            Language name. Ex: English.
+	 *     @type string  $slug            Language code used in URL. Ex: en.
+	 *     @type string  $locale          WordPress language locale. Ex: en_US.
+	 *     @type string  $w3c             W3C locale.
+	 *     @type string  $flag_code       Code of the flag.
+	 *     @type int     $term_group      Order of the language when displayed in a list of languages.
+	 *     @type int     $is_rtl          `1` if the language is rtl, `0` otherwise.
+	 *     @type int     $mo_id           Optional. ID of the post storing strings translations.
+	 *     @type string  $facebook        Optional. Facebook locale.
+	 *     @type string  $home_url        Optional. Home URL in this language.
+	 *     @type string  $search_url      Optional. Home URL to use in search forms.
+	 *     @type string  $host            Optional. Host corresponding to this language.
+	 *     @type string  $flag_url        Optional. URL of the flag.
+	 *     @type string  $flag            Optional. HTML markup of the flag.
+	 *     @type string  $custom_flag_url Optional. URL of the custom flag if it exists.
+	 *     @type string  $custom_flag     Optional. HTML markup of the custom flag if it exists.
+	 *     @type int     $page_on_front   Optional. ID of the page on front in this language.
+	 *     @type int     $page_for_posts  Optional. ID of the page for posts in this language.
+	 * }
 	 *
-	 * @param WP_Term|array $language      Term in 'language' taxonomy or language object properties stored as an array.
-	 * @param WP_Term       $term_language Corresponding 'term_language' term.
+	 * @phpstan-param array{
```

Comment:
`term_props` is mandatory since [2946a79](https://github.com/polylang/polylang/pull/1149/commits/2946a79f7e4892a88bd251b557e9e52c3e977eba) (not in the code thought, only in the doc), but I kept the old properties for PHPStan as optional, for backward compatibility. Note that they have been removed from the doc for humans.
Maybe we should remove them for PHPStan too indeed.

Reply:
I'll let you chose if we should remove them from PHPStan 😉